logger/logger.lua

```diff
@@ -0,0 +1,51 @@
+local string = require('string')
+local windower = require('windower')
+local files = require('files')
+local os = require('os')
+local chat = require('chat')
+local account = require('account')
+
+local get_log = function()
+    local dir = windower.user_path .. '\\' .. account.name .. '\\'
+    os.execute('mkdir "' .. dir .. '" >nul 2>nul')
+
+    local date = os.date('*t')
+    local file_timestamp = string.format('%.4u.%.2u.%.2u.log',date.year, date.month, date.day)
+    
+    return files.create(dir .. file_timestamp)
+end
+
+chat.text_added:register(function(obj)
+    if account.logged_in then
+        local file = get_log()
```

Comment:
The `get_log` function only really needs to run when the accounts change, otherwise it wastes a lot of effort.  Use `account.login:register(get_log)` and change `get_log` to set a global variable (can still be `local`), then use that in the `chat.text_added` event.

`account.logout:register` can register a function to reset the global variable to `nil`.

Reply:
What about when the day rolls over at midnight? I suppose I could add a date check every text_added callback to see if the day has changed since the initial opening and re run get_log.